contracts/KyberOasisReserve.sol

```diff
@@ -0,0 +1,273 @@
+pragma solidity 0.4.18;
+
+import "./ERC20Interface.sol";
+import "./Utils2.sol";
+import "./Withdrawable.sol";
+import "./KyberReserveInterface.sol";
+
+
+contract OtcInterface {
+    function getBuyAmount(ERC20 buyGem, ERC20 payGem, uint payAmt) public constant returns (uint fillAmt);
+    function sellAllAmount(ERC20 payGem, uint payAmt, ERC20 buyGem, uint minFillAmount) public returns (uint fillAmt);
+}
+
+
+contract TokenInterface is ERC20 {
+    function deposit() public payable;
+    function withdraw(uint) public;
+}
+
+
+contract KyberOasisReserve is KyberReserveInterface, Withdrawable, Utils2 {
+
+    uint constant internal ETH_SRC_AMOUNT = (10**18);
+    address public kyberNetwork;
+    OtcInterface public otc;
+    TokenInterface public wethToken;
+    ERC20 public tradeToken;
```

Comment:
shouldn't we just call it DAI token?

Reply:
let's leave it more generic, in case we want to have it for other tokens later (will have to have 18 decimals, and see if we can still use the MIN_TRADE_TOKEN_SRC_AMOUNT).